server/interpreter/testcases.js

```diff
@@ -0,0 +1,460 @@
+/**
+ * @license
+ * Code City interpreter JS testscases
+ *
+ * Copyright 2017 Google Inc.
+ * https://github.com/NeilFraser/CodeCity
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Test cases for JavaScript interpreter.
+ * @author cpcallen@google.com (Christopher Allen)
+ */
+
+// This file is input for (and required by) gentests.js.
+exports.tests = [
+  { name: "onePlusOne", src: `
+    1 + 1;
+    `,
+    expected: 2 },
+
+  { name: "twoPlusTwo", src: `
+    2 + 2;
+    `,
+    expected: 4 },
+
+  { name: "sixTimesSeven", src: `
+    6 * 7;
+    `,
+    expected: 42 },
+
+  { name: "simpleFourFunction", src: `
+    (3 + 12 / 4) * (10 - 3);
+    `,
+    expected: 42 },
+
+  { name: "variableDecl", src: `
+    var x = 43;
+    x;
+    `,
+    expected: 43 },
+
+  { name: "condTrue", src: `
+    true?"then":"else";
+    `,
+    expected: "then" },
+
+  { name: "condFalse", src: `
+    false?"then":"else";
+    `,
+    expected: "else" },
+
+  { name: "ifTrue", src: `
+    if(true) {
+        "then";
+    }
+    else {
+        "else";
+    }
+    `,
+    expected: "then" },
+
+  { name: "ifFalse", src: `
+    if(false) {
+        "then";
+    }
+    else {
```

Comment:
I'm getting used to cuddled elses in Go, but I still think this looks really ugly.

What puzzles me, though, is that I don't recall ever using them previously to this year, yet when I check the Google style guides for various languages I have used here (Java and Perl in particular) I see cuddling is apparently de rigueur.  Worse, looking up K&R style and 1TBS on Wikipedia it claims that both those styles cuddle their elses as well—a fact I am 100% sure isn't true.  Surely there has been some Wiki vandalism.  I even went and found a PDF of "The C Programming Language" to confirm that my memory is correct, but obviously the PDF has been tampered with because *it too contains cuddled elses*.

I am having a very [Berentst[ea]in Bears](https://www.vice.com/en_uk/article/the-berensteain-bears-conspiracy-theory-that-has-convinced-the-internet-there-are-parallel-universes) day.